src/components/filters/useOperatingSystemFilter.js

```diff
@@ -1,37 +1,88 @@
-import { useState } from 'react';
-import { buildOSFilterConfig, buildOSChip, toGroupSelectionValue } from '../../Utilities/OperatingSystemFilterHelpers';
+import get from 'lodash/get';
+import set from 'lodash/set';
+import { useCallback, useEffect, useMemo, useState } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import { fetchOperatingSystems } from '../../store/inventory-actions';
+import {
+    buildOSFilterChip,
+    getSelectedOsFilterVersions,
+    groupOSFilterVersions,
+    onOSFilterChange,
+    toGroupSelection
+} from '../../Utilities/OperatingSystemFilterHelpers';
 
-export const operatingSystemFilterState = { operatingSystemFilter: [] };
 export const OPERATING_SYSTEM_FILTER = 'OPERATING_SYSTEM_FILTER';
 export const operatingSystemFilterReducer = (_state, { type, payload }) => ({
     ...type === OPERATING_SYSTEM_FILTER && {
         operatingSystemFilter: payload
     }
 });
 
-export const useOperatingSystemFilter = (
-    operatingSystems,
-    areOperatingSystemsLoaded,
-    [state, dispatch] = [operatingSystemFilterState]
-) => {
-    // FIXME: is this state value needed and used at all?
-    const [operatingSystemStateValue, setStateValue] = useState([]);
-    const operatingSystemValue = dispatch ? state.operatingSystemFilter : operatingSystemStateValue;
-    const setValue = dispatch
-        ? (newValue) =>
-            dispatch({
-                type: OPERATING_SYSTEM_FILTER,
-                payload: Array.isArray(newValue)
-                    ? toGroupSelectionValue(newValue)
-                    : newValue
-            })
-        : setStateValue;
-    const filter = buildOSFilterConfig({
-        value: operatingSystemValue,
-        onChange: (_e, value) => setValue(value)
-        // TODO: better handling (spinner, skeleton) of loading systems?
-    }, areOperatingSystemsLoaded ? operatingSystems : []);
-    const chip = buildOSChip(operatingSystemValue,  areOperatingSystemsLoaded ? operatingSystems : []);
-
-    return [filter, chip, operatingSystemValue, setValue];
+/**
+* OS version filter hook.
+* @param {Array} apiParams - an array containing parameters for GET /system_profile/operating_system call
+* @return {Array} An array containing config object, chips array and value setter function.
+*/
+const useOperatingSystemFilter = (apiParams = []) => {
+    const dispatch = useDispatch();
+    const operatingSystems = useSelector(({ entities }) => entities?.operatingSystems);
+    const operatingSystemsLoaded = useSelector(({ entities }) => entities?.operatingSystemsLoaded) || false;
+    const availableVersions = operatingSystems === undefined ? [] : operatingSystems;
```

Comment:
Yeah, good point! We are just pulling `operatingSystems` from redux and rename them if found. If the `useSelector` doesn't find anything it returns undefined.

Reply:
@karelhala, @johnsonm325, this was done because one of `useEffect`s uses `operatingSystems` as a dependency. If, in redux store we forget to set `operatingSystems` default value to `[]` (the key doesn't exist, which can sometimes happen and we ideally have to cover such cases), this will lead for the hook to infinitely reload, because shallow comparison of two empty arrays equals to `false`. 

For example, I could see this while running tests: assume that we declare the variable with
```
const operatingSystems = useSelector(({ entities }) => entities?.operatingSystems) || [];
```
then the result of running tests will be the following

![изображение](https://user-images.githubusercontent.com/31385370/192524393-667d8cf0-2b45-4eb4-8bfd-338a9759aea6.png)

The reason is that one of the tests doesn't have `operatingSystems` mocked in its mock store. And yep, we just can update all the tests, but in the future, this will be really hard to understand whether the value in a store is optional or not.

Reply:
OK I managed to go another way: just wrapped with `|| []` check in places where the value can be undefined, but an array is expected. PTAL the last push